src/core/observables/Current.hpp

```diff
@@ -32,16 +32,13 @@ class Current : public PidObservable {
   std::vector<double>
   evaluate(Utils::Span<std::reference_wrapper<const Particle>> particles)
       const override {
-    std::vector<double> res(n_values());
+    Utils::Vector3d current{};
 #ifdef ELECTROSTATICS
     for (auto p : particles) {
-      double charge = p.get().p.q;
-      res[0] += charge * p.get().m.v[0];
-      res[1] += charge * p.get().m.v[1];
-      res[2] += charge * p.get().m.v[2];
+      current += (traits::charge(p) * traits::velocity(p));
     };
 #endif
-    return res;
+    return (Utils::hadamard_division(current, box_geo.length())).as_vector();
```

Comment:
I don't think that this is a good solution, because you don't know what the correct volume is. If you imagine particles confined to a cylinder across the box (e.g. a 'wire'), which this definition the current would change if you change the empty box around your wire. I think the solution is probably to just not call it current, but find a more correct name.

Reply:
changed it back, you are right